spec/README.md

```diff
@@ -0,0 +1,172 @@
+
+The smoothing pool aims to solve this problem, by creating a common pool to which all solo stakers give their rewards. Then, every given amount of time, these rewards are shared equally by all participants. This makes rewards more predictable for entities/solo-stakers running few validators, allowing them to average out or *smooth* the rewards they get. Note that entities running thousands of validators are not subject to this issue, because with that amount of validators statistics plays on their favor, so they internally smooth the variance.
+
+As an example, if `validator_a` proposes a block with `reward_a` and `validator_b` with `reward_b`, it is statistically better that instead of each getting `reward_a` and `reward_b` respectively, each one gets `(reward_a+reward_b)/2`. This oracle contains all the calculations of the rewards each participant validator can claim, for an arbitrary number of validators that keep joining and leaving the pool, with some even cheating. The oracle also summarizes the data in a merkle tree, where the root is stored onchain to keep all accounting decentralized and as open as possible.
+
+The pool has two main components:
+* **oracle**: Calculates off-chain the rewards that each participant of the smoothing pool can claim, using on-chain data from the consensus and execution layer. It also summarizes all balances and addresses with the right to claim in a merkle root, that is periodically stored on-chain, making it non-revertable. On the other hand it serves as a data availability layer, that can be used to get the proofs one needs to use on-chain to claim their share of the rewards.
+* **contract**: Gets all the rewards produced by the members of the pool and provides a set of functions to subscribe, unsubscribe and claim rewards by providing the appropriate proofs of belonging to the merkle tree.
+
+## Configuration parameters
+
+The oracle shall contain the following configuration parameters, that will determine how rewards are calculated:
+* `COLLATERAL_GWEI`: Amount of collateral in gwei that a validator need to join the pool.
+* `DEPLOYED_SLOT`: Slot when the smoothing pool contract was deployed.
+* `CHECKPOINT_SIZE_SLOTS`: How often the smart contract root is updated with new rewards, in slots.
+* `OWNER_ADDRESS`: Address with `0x` prefix of the account allowed to update the smoothing pool contract.
+* `POOL_CONTRACT_ADDRESS`: Address with `0x` prefix of the smoothing pool contract.
+* `NETWORK`: Network where rewards are being calculated: `mainnet` or `goerli`.
+* `POOL_FEES_ADDRESS`: Address with `0x` prefix of the account that can claim the smoothing pool fees.
+* `POOL_FEES_PERCENT`: Amount in % that `POOL_FEES_ADDRESS` gets for every reward sent to the smoothing pool. Note that it also gets rounding remainders on top, but this is almost neglectable.
+
```

Comment:
self note. include notes on reconciliation process:
-owe: total pending+accum+fees. minus already claimed for all.
-have: pool balance

note: run all numbers only in finalaized epochs.
also add this to spec. only finalized epochs!